### PR TITLE
[fix](agg) crash caused by failure of prepare

### PR DESCRIPTION
**Pull request by mrhhsg**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/vaggregation_node.cpp [hunk 1]
```diff
@@ -499,11 +508,14 @@ Status AggregationNode::close(RuntimeState* state) {
     VExpr::close(_probe_expr_ctxs, state);
     if (_executor.close) _executor.close();
 
-    std::visit(
-            [&](auto&& agg_method) {
-                COUNTER_SET(_hash_table_size_counter, int64_t(agg_method.data.size()));
-            },
-            _agg_data._aggregated_method_variant);
+    /// _hash_table_size_counter may be null if prepare failed.
+    if (_hash_table_size_counter) {
```
Comment (Gabriel39) [inline be/src/vec/exec/vaggregation_node.cpp:512]: If prepare failed, `close` should be never called.

Comment (mrhhsg) [reply be/src/vec/exec/vaggregation_node.cpp:512]: Actually in function `Status FragmentMgr::exec_plan_fragment(const TExecPlanFragmentParams& params, FinishCallback cb)` when prepare was failed, the destructor of `PlanFragmentExecutor` will call `close`.

Comment (Gabriel39) [inline be/src/vec/exec/vaggregation_node.cpp:512]:
```suggestion
    if (LIKELY(_hash_table_size_counter)) {
```

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**